api/lib.go

```diff
@@ -232,6 +232,234 @@ func Query(
 	return receiveVector(res), uint64(gasUsed), nil
 }
 
+func IBCChannelOpen(
+	cache Cache,
+	code_id []byte,
+	env []byte,
+	channel []byte,
+	gasMeter *GasMeter,
+	store KVStore,
+	api *GoAPI,
+	querier *Querier,
+	gasLimit uint64,
+	printDebug bool,
+) ([]byte, uint64, error) {
+	id := sendSlice(code_id)
+	defer freeAfterSend(id)
+	e := sendSlice(env)
+	defer freeAfterSend(e)
+	c := sendSlice(channel)
+	defer freeAfterSend(c)
+
+	// set up a new stack frame to handle iterators
+	counter := startContract()
+	defer endContract(counter)
+
+	dbState := buildDBState(store, counter)
+	db := buildDB(&dbState, gasMeter)
+	a := buildAPI(api)
+	q := buildQuerier(querier)
+	var gasUsed cu64
+	errmsg := C.Buffer{}
+
+	res, err := C.ibc_channel_open(cache.ptr, id, e, c, db, a, q, cu64(gasLimit), cbool(printDebug), &gasUsed, &errmsg)
```

Comment:
I have noticed that the setup looks very similar in all the IBC callbacks. Would it make sense to refactored them into the command patten? (in a new PR)

Reply:
I did that for the rust side on receiving the entry points.
I would be happy if you want to make the Go side a bit drier.

Reply:
👍  Would be happy to do. I will open an issue.